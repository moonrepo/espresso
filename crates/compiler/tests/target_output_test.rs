mod utils;

use espresso_common::EsTarget;
use espresso_compiler::Compiler;
use espresso_package::Package;
use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::read_file;

macro_rules! test_target {
    ($method:ident, $target:expr) => {
        #[tokio::test]
        async fn $method() {
            let sandbox = create_sandbox("syntax");
            let package = Package::new(sandbox.path()).unwrap();
            let compiler = Compiler::new(&package).unwrap();
            let out_dir = compiler.compile($target).await.unwrap();

            assert_snapshot!(read_file(out_dir.join("index.mjs")));
        }
    };
}

mod target_output {
    use super::*;

    test_target!(es2015, EsTarget::Es2015);
    test_target!(es2016, EsTarget::Es2016);
    test_target!(es2017, EsTarget::Es2017);
    test_target!(es2018, EsTarget::Es2018);
    test_target!(es2019, EsTarget::Es2019);
    test_target!(es2020, EsTarget::Es2020);
    test_target!(es2021, EsTarget::Es2021);
    test_target!(es2022, EsTarget::Es2022);

    #[tokio::test]
    async fn supports_legacy_decorators() {
        let sandbox = create_sandbox("syntax-legacy-decorators");
        let package = Package::new(sandbox.path()).unwrap();
        let compiler = Compiler::new(&package).unwrap();
        let out_dir = compiler.compile(EsTarget::Es2018).await.unwrap();

        assert_snapshot!(read_file(out_dir.join("index.mjs")));
        assert_snapshot!(read_file(out_dir.join("other.mjs")));
    }
}
