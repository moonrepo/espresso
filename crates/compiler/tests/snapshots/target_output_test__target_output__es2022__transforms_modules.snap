---
source: crates/compiler/tests/target_output_test.rs
expression: "read_file(out_dir.join(\"index.mjs\"))"
---
export class ClassName {
    field = 123;
    #private = true;
    method() {
        this.#private = false;
    }
    anonMethod = ()=>{
        this.field = 456_789;
        console.log(this.#private);
    };
    static field = 'abc';
    static{
        this.field = 'xyz';
    }
}
export async function asyncFunc(...args) {
    return Promise.resolve();
}
export function* genFunc(value = 'abc') {
    yield 1;
}
export async function* asyncGenFunc() {
    yield 'abc';
}
const obj = {
    a: 1,
    b: 2,
    c: 3
};
const { a, b, c } = obj;
const [d, e, f] = [
    1,
    2,
    3
];
const nullable = 1 ?? 2 ?? 3;
const optional = obj?.a || 0;
let assign = 0;
assign ||= a;
assign &&= b;
assign ??= b; // Note: Can't test React because we don't have node modules!
 // export function Component() {
 // 	return <div />;
 // }

